youtube_dl/extractor/gedidigital.py

```diff
@@ -0,0 +1,170 @@
+        'only_matching': True,
+    }, {
+        'url': 'https://video.ilpiccolo.gelocal.it/sport/dentro-la-notizia-ferrari-cosa-succede-a-maranello/133362/134466',
+        'only_matching': True,
+    }, {
+        'url': 'https://video.gazzettadimantova.gelocal.it/sport/dentro-la-notizia-ferrari-cosa-succede-a-maranello/133362/134466',
+        'only_matching': True,
+    }, {
+        'url': 'https://video.mattinopadova.gelocal.it/sport/dentro-la-notizia-ferrari-cosa-succede-a-maranello/133362/134466',
+        'only_matching': True,
+    }, {
+        'url': 'https://video.laprovinciapavese.gelocal.it/sport/dentro-la-notizia-ferrari-cosa-succede-a-maranello/133362/134466',
+        'only_matching': True,
+    }, {
+        'url': 'https://video.tribunatreviso.gelocal.it/sport/dentro-la-notizia-ferrari-cosa-succede-a-maranello/133362/134466',
+        'only_matching': True,
+    }, {
+        'url': 'https://video.nuovavenezia.gelocal.it/sport/dentro-la-notizia-ferrari-cosa-succede-a-maranello/133362/134466',
+        'only_matching': True,
+    }, {
+        'url': 'https://video.gazzettadimodena.gelocal.it/sport/dentro-la-notizia-ferrari-cosa-succede-a-maranello/133362/134466',
+        'only_matching': True,
+    }, {
+        'url': 'https://video.lanuovaferrara.gelocal.it/sport/dentro-la-notizia-ferrari-cosa-succede-a-maranello/133362/134466',
+        'only_matching': True,
+    }, {
+        'url': 'https://video.corrierealpi.gelocal.it/sport/dentro-la-notizia-ferrari-cosa-succede-a-maranello/133362/134466',
+        'only_matching': True,
+    }, {
+        'url': 'https://video.lasentinella.gelocal.it/sport/dentro-la-notizia-ferrari-cosa-succede-a-maranello/133362/134466',
+        'only_matching': True,
+    }]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+
+        webpage = self._download_webpage(url, video_id)
+        player_data = re.findall(
+            r'''PlayerFactory\.setParam\('(?P<type>format|param)',\s*'(?P<name>(?:audio|video|image).*?)',\s*'(?P<val>.+?)'\);''',
+            webpage)
+
+        formats = []
+        thumb = None
+
+        for t, n, v in player_data:
+            if t == 'format':
+                if n == 'video-hls-vod-ak':
+                    formats.extend(self._extract_akamai_formats(
+                        v, video_id, {'http': 'media.gedidigital.it'}))
+                if n == 'audio-hls-vod':
+                    formats.extend(self._extract_m3u8_formats(
+                        v, video_id, 'm4a', m3u8_id='audio-hls', fatal=False))
+                if n == 'audio-rrtv-mp3':
+                    formats.append({
+                        'format_id': 'audio-mp3',
+                        'url': v,
+                        'tbr': 128,
+                        'ext': 'mp3',
+                        'vcodec': 'none',
+                        'acodec': 'mp3',
+                    })
```

Comment:
again, extract all formats, there is no guarentee that those formats are available in all videos.

Reply:
`video-hls-vod-ak` is always present and from that with _extarct_akamai it creates formats for hls, hds and http/https;
I think it's a cleaner solution that to add stuff from the website (like in some page there is also a youtube link of the same video as `video-youtube-pfp`) so I didn't want to pollute the format list with duplicates etc.

Reply:
as i said there is no guarantee that this will be the case all the time.